File: android/app/src/main/java/com/example/reactnativerainbowmodule/MyFirebaseService.java
package com.example.reactnativerainbowmodule;
import android.os.Build;
import android.util.Log;

import androidx.annotation.NonNull;
import androidx.annotation.RequiresApi;


import com.ale.rainbow.rn.notifications.NotificationHandler;
import com.google.firebase.messaging.FirebaseMessagingService;
import com.google.firebase.messaging.RemoteMessage;

public class MyFirebaseService  extends FirebaseMessagingService{
    private NotificationHandler notificationHandler;

    @Override
    public void onCreate() {
        super.onCreate();
        notificationHandler = new NotificationHandler(this);
        notificationHandler.setChannelId(MainActivity.CHANNEL_ID);
    }
    @Override
    public void onNewToken(@NonNull String refreshedToken) {
        super.onNewToken(refreshedToken);
        Log.i("MyFirebaseService", "onNewToken: "+refreshedToken);
        notificationHandler.onTokenRefresh(refreshedToken);
    }

    @RequiresApi(api = Build.VERSION_CODES.O)
    @Override
    public void onMessageReceived(@NonNull RemoteMessage remoteMessage) {
        Log.i("MyFirebaseService", "onMessageReceived: "+remoteMessage);
        notificationHandler.onPushMessageReceived(remoteMessage,this.getApplication(),MainActivity.class);
        super.onMessageReceived(remoteMessage);
    }
}

File: android/app/src/main/java/com/example/reactnativerainbowmodule/MyFirebaseService.kt
package com.example.reactnativerainbowmodule
import android.util.Log
import com.ale.rainbow.rn.notifications.NotificationHandler
import com.ale.rainbow.rn.notifications.NotificationHandler.Companion.setChannelId
import com.google.firebase.messaging.FirebaseMessagingService
import com.google.firebase.messaging.RemoteMessage

class MyFirebaseService : FirebaseMessagingService() {
    private var notificationHandler: NotificationHandler? = null

    override fun onCreate() {
        super.onCreate()
        notificationHandler = NotificationHandler(this)
        setChannelId(MainActivity.CHANNEL_ID)
    }

    override fun onNewToken(refreshedToken: String) {
        super.onNewToken(refreshedToken)
        Log.i(TAG, "onNewToken: $refreshedToken")
        notificationHandler!!.onTokenRefresh(refreshedToken)
    }

    override fun onMessageReceived(remoteMessage: RemoteMessage) {
        Log.i(TAG, "onMessageReceived: $remoteMessage")
        notificationHandler!!.onPushMessageReceived(
            remoteMessage,
            this.application,
            MainActivity::class.java
        )
        super.onMessageReceived(remoteMessage)
    }
    companion object {
        private val TAG: String = MyFirebaseService::class.java.simpleName
    }
}
